freqtrade/rpc/external_message_consumer.py

```diff
@@ -0,0 +1,332 @@
+"""
+ExternalMessageConsumer module
+
+Main purpose is to connect to external bot's message websocket to consume data
+from it
+"""
+import asyncio
+import logging
+import socket
+from threading import Thread
+from typing import TYPE_CHECKING, Any, Callable, Dict, List
+
+import websockets
+from pydantic import ValidationError
+
+from freqtrade.data.dataprovider import DataProvider
+from freqtrade.enums import RPCMessageType
+from freqtrade.exceptions import OperationalException
+from freqtrade.misc import remove_entry_exit_signals
+from freqtrade.rpc.api_server.ws import WebSocketChannel
+from freqtrade.rpc.api_server.ws_schemas import (WSAnalyzedDFMessage, WSAnalyzedDFRequest,
+                                                 WSMessageSchema, WSRequestSchema,
+                                                 WSSubscribeRequest, WSWhitelistMessage,
+                                                 WSWhitelistRequest)
+
+
+if TYPE_CHECKING:
+    import websockets.connect
+    import websockets.exceptions
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalMessageConsumer:
+    """
+    The main controller class for consuming external messages from
+    other FreqTrade bot's
+    """
+
+    def __init__(
+        self,
+        config: Dict[str, Any],
+        dataprovider: DataProvider
+    ):
+        self._config = config
+        self._dp = dataprovider
+
+        self._running = False
+        self._thread = None
+        self._loop = None
+        self._main_task = None
+        self._sub_tasks = None
+
+        self._emc_config = self._config.get('external_message_consumer', {})
+
+        self.enabled = self._emc_config.get('enabled', False)
+        self.producers = self._emc_config.get('producers', [])
+
+        if self.enabled and len(self.producers) < 1:
+            raise OperationalException("You must specify at least 1 Producer to connect to.")
+
+        self.wait_timeout = self._emc_config.get('wait_timeout', 300)  # in seconds
+        self.ping_timeout = self._emc_config.get('ping_timeout', 10)  # in seconds
+        self.sleep_time = self._emc_config.get('sleep_time', 10)  # in seconds
+
+        # The amount of candles per dataframe on the initial request
+        self.initial_candle_limit = self._emc_config.get('initial_candle_limit', 1500)
+
+        # Message size limit, in megabytes. Default 8mb, Use bitwise operator << 20 to convert
+        # as the websockets client expects bytes.
+        self.message_size_limit = (self._emc_config.get('message_size_limit', 8) << 20)
```

Comment:
i think this should be aligned with the value in the schema in constants (which is using `max_message_size`).

now which way we're going to fix this is not relevant - but the two should be aligned for sure.

Reply:
Yes it should actually be 'message_size_limit', must've been a long day.